src/libraries/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs

```diff
@@ -176,6 +176,7 @@ public sealed partial class CoClassAttribute : System.Attribute
     public static partial class CollectionsMarshal
     {
         public static System.Span<T> AsSpan<T>(System.Collections.Generic.List<T>? list) { throw null; }
+        public static ref TValue GetValueRefOrNullRef<TKey, TValue>(System.Collections.Generic.Dictionary<TKey, TValue> dictionary, TKey key) where TKey : notnull { throw null; }
```

Comment:
Shouldn't we name this `GetValueRef` if we're not planning on shipping a throwing variant?

Reply:
Would be unclear what happens if it doesn't exist? Though the other variant would be ```GetValueRefOrAddDefault```

Reply:
Even if we're not shipping a throwing variant, I think someone could reasonably assume that this would throw, and the "OrNullRef" helps to clarify it doesn't, while also highlighting what the dev needs to compare against.